pkg/testcase/conformance.go

```diff
@@ -0,0 +1,137 @@
+	testResultTar, err := shared.RunCommandHost(cmd)
+	Expect(err).NotTo(HaveOccurred(), "failed cmd: "+cmd)
+
+	cmd = "sonobuoy results  " + testResultTar
+	res, err = shared.RunCommandHost(cmd)
+	Expect(err).NotTo(HaveOccurred(), "failed cmd: "+cmd)
+	Expect(res).Should(ContainSubstring("Plugin: mixed-workload-e2e\nStatus: passed\n"))
+
+	if deleteWorkload {
+		cmd = "sonobuoy delete --all --wait --kubeconfig=" + shared.KubeConfigFile
+		_, err = shared.RunCommandHost(cmd)
+		Expect(err).NotTo(HaveOccurred(), "failed cmd: "+cmd)
+		sonobuoyVersion := customflag.ServiceFlag.External.SonobuoyVersion
+		err = shared.InstallSonobuoy("delete", sonobuoyVersion)
+		if err != nil {
+			GinkgoT().Errorf("error: %v", err)
+			return
+		}
+	}
+}
+
+func ConformanceTest(cluster *shared.Cluster) {
+	verifyClusterNodes(cluster)
+	installConformanceBinary()
+	launchSonobuoyTests("certified-conformance")
+	// refactor later to accept quick or certified-conformance
+	testResultTar := checkStatusGetResults(cluster)
+	shared.LogLevel("info", "%s", "testResultTar: "+testResultTar)
+	// need to do cilium force failures to test
+	rerunFailedTests(testResultTar)
+	parseResults(testResultTar)
+	cleanupTests()
+}
+
+func verifyClusterNodes(cluster *shared.Cluster) bool {
+	if cluster.NumAgents < 1 && cluster.NumServers < 1 {
+		shared.LogLevel("error", "%s", "cluster does not meet the minimum requirements to run conformance tests")
+		return false
+	}
+
+	return true
+}
+
+func installConformanceBinary() {
+	sonobuoyVersion := customflag.ServiceFlag.External.SonobuoyVersion
+	err := shared.InstallSonobuoy("install", sonobuoyVersion)
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func launchSonobuoyTests(testMode string) {
+	// not doing anything different yet if the status is running from the previous attempts
+	cmds := "kubectl get namespace sonobuoy --kubeconfig=" + shared.KubeConfigFile
+	res, _ := shared.RunCommandHost(cmds)
+	if strings.Contains(res, "Active") {
+		shared.LogLevel("info", "%s", "sonobuoy namespace is active, waiting for it to complete")
+		return
+	}
+	if strings.Contains(res, "Error from server (NotFound): namespaces \"sonobuoy\" not found") {
+		cmd := "sonobuoy run --kubeconfig=" + shared.KubeConfigFile +
+			" --mode=" + testMode + " --kubernetes-version=" + shared.ExtractKubeImageVersion()
+		_, err := shared.RunCommandHost(cmd)
+		Expect(err).NotTo(HaveOccurred())
+	}
+}
+
+func checkStatusGetResults(cluster *shared.Cluster) string {
+	// sonobuoy's output is becoming unreliable for status checks observe remaining count incorrect at 404
+	// 	sono status
+	//          PLUGIN     STATUS   RESULT   COUNT                                PROGRESS
+	//             e2e   complete   passed       1   Passed:  0, Failed:  0, Remaining:404
+	//    systemd-logs   complete   passed       2
+	// Sonobuoy has completed. Use `sonobuoy retrieve` to get results
+	cmd := "sonobuoy status --kubeconfig=" + shared.KubeConfigFile
```

Comment:
I think we can separate the status check and get results in to their own func so that it is simpler to understand the error returned in case of failures